src/main/scala/org/clulab/wm/eidos/attachments/NegationHandler.scala

```diff
@@ -175,11 +174,14 @@ class NegationHandler(val language: String) {
 
     // Get the token interval of the event, but exclude the intervals of the arguments
     val argumentIntervals = event.arguments.values.flatten.map(_.tokenInterval)
+    val sentenceWords = event.sentenceObj.words
     // Check for single-token negative verbs
     for {
       (ix, lemma) <- leftContext ++ rightContext
       if !argumentIntervals.exists(_.contains(ix))
-      if (failNot contains lemma) && !(previouslyFound contains ix) && event.sentenceObj.words(ix+1) != "only"
+      // make sure ix+1 is not the end of the sentence (to avoid ix+1 error at end of sent)
+      if rightContext.isEmpty || (rightContext.nonEmpty && rightContext.last._1 < sentenceWords.lastIndexOf(sentenceWords.last))
+      if (NegationHandler.failNot contains lemma) && !(previouslyFound contains ix) && event.sentenceObj.words(ix+1) != "only"
```

Comment:
For someone like me who is not as familiar with the code, it's quite difficult to follow the relationship between rightContext and ix.  I'm hoping for an option that will minimize the probability that I (or others) will break this later, and that's a nearly impossible task :-(  Your patience is appreciated!

This version passes for me, but if it is wrong or it doesn't make sense to you, please "improve" it as needed:

```scala
    for {
      (ix, lemma) <- leftContext ++ rightContext
      // These are ordered roughly from fastest and most effective to slowest and least.
      if NegationHandler.failNot contains lemma     // lemma is either "fail" or "not", and
      if !(previouslyFound contains ix)             // ix is not a duplicate, and
      if !argumentIntervals.exists(_.contains(ix))  // ix is not part of any argument interval, and
      if !sentenceWords.indices.contains(ix + 1) || // either there is no next word at all or
          sentenceWords(ix + 1) != "only"           // there is one and the next word is not "only"
    } yield new TextBoundMention(
```
That is definitely a complicated set of circumstances.